cmu_graphics/cmu_graphics.py

```diff
@@ -387,7 +412,83 @@ def call_updater_with_args(args):
         pass
 
 def loop():
-    app.run()
+    app._app.run()
+
+from code import InteractiveConsole
+class CSAcademyConsole(InteractiveConsole):
+    def __init__(self):
+        self.name = "CS Academy Console"
+        __main__.__dict__['exit'] = lambda: os._exit(0)
+        super().__init__(locals=__main__.__dict__, filename = '<%s>' % self.name)
+
+    # Override the default error handling functions to avoid using our own
+    # excepthook function
+    def showsyntaxerror(self, filename=None):
+        type, value, tb = sys.exc_info()
+        sys.last_type = type
+        sys.last_value = value
+        sys.last_traceback = tb
+        if filename and type is SyntaxError:
+            # Work hard to stuff the correct filename in the exception
+            try:
+                msg, (dummy_filename, lineno, offset, line) = value.args
+            except ValueError:
+                # Not the format we expect; leave it alone
+                pass
+            else:
+                # Stuff in the right filename
+                value = SyntaxError(msg, (filename, lineno, offset, line))
+                sys.last_value = value
+
+        lines = traceback.format_exception_only(type, value)
+        self.write(''.join(lines))
+
+    def showtraceback(self):
+        sys.last_type, sys.last_value, last_tb = ei = sys.exc_info()
+        sys.last_traceback = last_tb
+        try:
+            lines = traceback.format_exception(ei[0], ei[1], last_tb.tb_next)
+            self.write(''.join(lines))
+        finally:
+            last_tb = ei = None
+
+    # Override interact so we can os._exit on EOF
```

Comment:
This code seems to work correctly as far as exiting on EOF:

```python
    def interact(self):
        cprt = 'Type "help", "copyright", "credits" or "license" for more information.'

        InteractiveConsole.interact(self, banner="Python %s on %s\n%s\n(%s)\n" %
                   (sys.version, sys.platform, cprt,
                    self.name))

        os._exit(0)
```

but it doesn't do your improved handling of the KeyboardInterrupt exception. I like it because there is less copy/pasted code, but I also like the cleaner handling of KeyboardInterrupt. Your call on what to do, though it would be good to update the comment.

Reply:
I think I'd initially rewritten the whole thing because I wanted to change or suppress the exit message. The KeyboardInterrupt handler is there already at leas in the Python 3.9 source. 

The newest commit overrides `self.__class__.__name__` so the banner and exit messages are nice, uses `super().interact()` so there's less copy pasted code, and also wraps `app.run()` in a try/except block for KeyboardInterrupt handling